### PR TITLE
Slugified Anchor Links

### PR DESCRIPTION
**Pull request by archiewood**
### Description

Addresses #1352

- Adds automatic anchor links to headings (H1-H6) in the preprocess stage.
- Puts H1-H3 in the TOC (rather than H1-H2)
- Links are based on slugified forms of the headers, but are made unique in the case of duplication by appending numbers
- Allows external links to navigate to the correct section of the page eg: `my-project.evidence.app/page-1/#heading-on-the-page`

### Issues
- [ ] Issue (svelte kit bug? https://github.com/sveltejs/kit/issues/11326) which prevents navigation between pages in Evidence from working correctly. **Open issue for this**
  - Appears to be client side nav only issue
  - This is not a regression as this did not work before.
- This uses outdated versions of rehype packages that still support CJS. I suspect the long term solution here is to convert this whole module to ESM, but then we need to rip out Prism (code highlighting) which is CJS only.

CleanShot 2024-03-27 at 22 15 54

### Checklist

- [ ] For UI or styling changes, I have added a screenshot or gif showing before & after
- [x] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)
- [ ] I have added to the docs where applicable
- [ ] I have added to the [VS Code extension](https://github.com/evidence-dev/evidence-vscode) where applicable

### CODE REVIEW

File: packages/ui/core-components/src/lib/organisms/layout/tableofcontents/ContentsList.svelte [hunk 1]
```diff
@@ -38,12 +34,8 @@
...
 		On this page
 	</span>
-	{#each headers as header, i}
-		<a
-			href={'#' + encodeURIComponent(i + 1)}
-			class={header.nodeName.toLowerCase()}
-			class:first={i === 0}
-		>
+	{#each headers as header}
+		<a href={'#' + header.id} class={header.nodeName.toLowerCase()}>
```
Comment (csjh) [inline packages/ui/core-components/src/lib/organisms/layout/tableofcontents/ContentsList.svelte:38]:
```suggestion
		<a href="#{header.id}" class={header.nodeName.toLowerCase()}>
```

**Review by csjh — APPROVED** [review]

File: sites/example-project/src/pages/links-to-page-headers/+page.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+
+Doesn't currently work
+
+[Deep in the text](/text-and-metrics/text#more-nostra-non-parsque-talia-discussit)
```
Comment (mcrascal) [inline sites/example-project/src/pages/links-to-page-headers/+page.md:4]: Could we open an issue instead of doing this?

**Review by mcrascal — APPROVED** [review]

File: sites/example-project/src/pages/links-to-page-headers/+page.md (hunk 2 above)
Comment (archiewood) [reply sites/example-project/src/pages/links-to-page-headers/+page.md:4]: good idea, will do upon merge

Comment (archiewood) [reply sites/example-project/src/pages/links-to-page-headers/+page.md]: #1871

**Pull request merged**